berkeley-mobile/Common/Images/ImageLoader.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ImageLoader.swift
+//  berkeley-mobile
+//
+//  Created by Shawn Huang on 10/10/20.
+//  Copyright © 2020 ASUC OCTO. All rights reserved.
+//
+
+import UIKit
+
+/// Loads images and keeps tracks of all previously loaded images by url, all current tasks by uuid
+class ImageLoader {
+    /// Shared ImageLoader instance to be used across the app
+    public static var shared = ImageLoader()
+    /// All images are cached here, all objects fetch cached images from here by url
+    private var loadedImages = [URL: UIImage]()
+    private var runningRequests = [UUID: URLSessionDataTask]()
+    
+    /// Load the image with specified url, run the completion once loaded, return the uuid for the task to load the image
+    @discardableResult
+    func getImage(url: URL, completion: @escaping (Result<UIImage, Error>) -> Void) -> UUID? {
+        if let image = loadedImages[url] {
+            completion(.success(image))
+            return nil
+        }
+        let uuid = UUID()
+        let task = URLSession.shared.dataTask(with: url) { data, response, error in
+            defer { self.runningRequests.removeValue(forKey: uuid) }
+            if let data = data, let image = UIImage(data: data) {
+                self.loadedImages[url] = image
+                completion(.success(image))
+                return
+            }
+            guard let error = error else { return }
+            // If task wasn't cancelled and we got another error, report that the load failed
+            guard (error as NSError).code == NSURLErrorCancelled else {
```

Comment:
Hmm, I'm still getting a lot of output in the console.
<img width="857" alt="Screen Shot 2020-10-11 at 11 05 42 PM" src="https://user-images.githubusercontent.com/41145903/95710849-6946e600-0c16-11eb-9013-a1de9b55bba1.png">

Is there any way to make this less verbose?

Reply:
yeah we can simply not print the error (this is done in the completion handlers on failure right now) or we can check for this specific error code. can't really test this rn because I'm having issues with GoogleService-Info.plist and so I can't see these errors, but should be simple fix